db/migrate/20170301150018_delete_orphaned_data.rb

```diff
@@ -0,0 +1,48 @@
+class DeleteOrphanedData < ActiveRecord::Migration
+  def change
+    #delete works for which there is no collection
+    Work.all.each {|w| w.destroy unless w.collection}
+    
+    #delete pages for which there are no works
+    Page.all.each {|p| p.destroy unless p.work}
+    
+    #delete document sets for which there is no collection
+    DocumentSet.all.each {|d| d.destroy unless d.collection}
+
+    #delete notes for which there is no collection
+    Note.all.each {|n| n.destroy unless n.collection}
```

Comment:
My logic was that a note should be deleted if it belongs to a collection which doesn't exist, because you can't have works or pages (or anything, really) on a collection that doesn't exist.